saga/saga.go

```diff
@@ -196,20 +213,25 @@ type sagaUpdate struct {
 // if msg is an invalid transition, it will neither log nor update internal state
 // always returns the new SagaState that should be used, either the mutated one or a copy of the
 // original.
-//
-func logMessage(state *SagaState, msg SagaMessage, log SagaLog) (*SagaState, error) {
-
+func logMessages(state *SagaState, msgs []SagaMessage, log SagaLog) (*SagaState, error) {
 	// updateSagaState will mutate state if it's a valid transition, but if we then error storing,
 	// we'll need to revert to the old state.
 	oldState := copySagaState(state)
-	//verify that the applied message results in a valid state
-	err := updateSagaState(state, msg)
-	if err != nil {
-		return oldState, err
+	// verify that the applied message results in a valid state
+	var err error
+	for _, msg := range msgs {
+		err = updateSagaState(state, msg)
+		if err != nil {
+			return oldState, err
+		}
 	}
 
-	//try durably storing the message
-	err = log.LogMessage(msg)
+	// try durably storing the message
+	if len(msgs) == 1 {
+		err = log.LogMessage(msgs[0])
```

Comment:
just using LogBatchMessages breaks like 20 million mock tests and I am refusing to fix them

Reply:
can we just remove the call to log.LogMessage() - we aren't really doing an durable store anyway.

Reply:
I don't follow, we are doing a durable store here.

Reply:
I just spent a while trying to consolidate the mock based tests and it's too painful, deferring it to wider refactor